### PR TITLE
Adds Owen's T function

### PR DESCRIPTION
**Pull request by ValentinKaisermayer**
```julia
    using Plots, SpecialFunctions
    plot(-3:0.1:3, -3:0.1:3, owens_t, st=:surface)
```
OwensT

```julia
julia> all(owens_t.(-3:3, 0) .== 0.)
true

julia> all(isapprox.(owens_t.(0, -3:3), 1 ./ (2 .* pi) .* atan.(-3:3), rtol=1e-6))
true

julia> all(isapprox.([owens_t(-h, a) for h in -3:3, a in -3:3], [owens_t(h, a) for h in -3:3, a in -3:3], rtol=1e-6))
true

julia> all(isapprox.([owens_t(h, -a) for h in -3:3, a in -3:3], [-owens_t(h, a) for h in -3:3, a in -3:3], rtol=1e-6))
true
```

### CODE REVIEW

**Comment by ohmsweetohm1** [comment]
Ported from http://people.sc.fsu.edu/~jburkardt/m_src/asa005/tfn.m

**Comment by giordano** [comment]
Note that that file has the following license:

> This code is distributed under the GNU LGPL license.

while this package has [MIT License](https://github.com/JuliaMath/SpecialFunctions.jl/blob/1bb0913fc667d6aebbc2e29fc111cd3cb8629771/LICENSE).  As far as I know, LGPL is compatible with MIT license, but you _must_ attribute the work to the original author and use the same license for this specific function, and I believe it'd be better to make it explicit in the license of this package that this function has a different license.

File: src/owens_t.jl [hunk 1]
```diff
@@ -0,0 +1,63 @@
+"""
+    owens_t(x, a)
+
+Owen's T function
+
+Ported from Matlab implementation of John Burkardt, which is distributed under the GNU LGPL license.
```
Comment (stevengj) [inline src/owens_t.jl:6]: This is more restrictive than the SpecialFunctions.jl license, which also disqualifies anything derived from that code.

**Comment by stevengj** [comment]
>  As far as I know, LGPL is compatible with MIT license

It's compatible in that you can legally comply with both licenses simultaneously, but it would effectively put SpecialFunctions.jl under the LGPL (since the terms of the LGPL are more restrictive than MIT, and apply to all derived works including translations to other languages).

**Comment by ohmsweetohm1** [comment]
[Scipy](https://github.com/scipy/scipy/blob/8dba340293fe20e62e173bdf2c10ae208286692f/scipy/special/cephes/owens_t.c) implements a version from [Boost](https://www.boost.org/doc/libs/1_67_0/libs/math/doc/html/math_toolkit/owens_t.html).
Which in turn is based on the paper [Patefield et al. - 2000 - Fast and Accurate Calculation of Owen's T Function](https://www.researchgate.net/publication/5142787_Fast_and_Accurate_Calculation_of_Owen's_T_Function).

**Comment by azev77** [comment]
@ohmsweetohm1 this would be really nice, as it would allow a CDF for SkewNormal in Distributions.jl

**Comment by ValentinKaisermayer** [comment]
Closing this, due to licencing issues

**Pull request closed without merging**